src/sknnr_spatial/estimator.py

```diff
@@ -230,12 +253,18 @@ def kneighbors(
         neigh_ind : Numpy or Xarray image with 3 dimensions (y, x, neighbor)
             Indices of the nearest points in the population matrix.
         """
-        wrapper = get_image_wrapper(X_image)(X_image, nodata_vals=nodata_vals)
-        self._check_feature_names(wrapper.preprocessor.band_names)
-
-        return wrapper.kneighbors(
-            estimator=self,
-            n_neighbors=n_neighbors,
+        image = Image.from_image(X_image, nodata_vals=nodata_vals)
+        k = n_neighbors or cast(int, getattr(self._wrapped, "n_neighbors", 5))
+
+        self._check_feature_names(image.band_names)
+
+        return image.apply_ufunc_across_bands(
+            suppress_feature_name_warnings(self._wrapped.kneighbors),
+            output_dims=[["k"], ["k"]] if return_distance else [["k"]],
```

Comment:
Our `kneighbors` used to return a dimension `variable=["k1", "k2", ...]`, but I switched this here to `k=[1, 2, ...]`. With DataArrays this seems clearer to me since you can do `nn.sel(k=1)` instead of `nn.sel(variable="k1")`, but with Datasets it's slightly weird because you have to do `nn[1]` instead of `nn.k1`.

Any thoughts or preferences on that decision?

Reply:
I like the `k=[1, 2, ...]` dimension naming better and can get used to the indexing on Datasets.